online-shop/src/app/services/shopping-cart.service.ts

```diff
@@ -0,0 +1,62 @@
+import {Injectable} from '@angular/core';
+import {Product} from '../models/product';
+import {ShoppingCartItem} from '../models/shopping-cart-item';
+import {HttpClient, HttpHeaders} from '@angular/common/http';
+import {Observable} from 'rxjs';
+import {OrderInput} from '../models/order-input';
+import {AppConfig} from '../app.config';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ShoppingCartService {
+  private shoppingCart: ShoppingCartItem[] = [];
+  private ordersUrl = AppConfig.API_ENDPOINT + '/orders';
+  private httpOptions = {
+    headers: new HttpHeaders({
+      'Content-Type': 'application/json'
+    })
+  };
+
+  constructor(private http: HttpClient) {
+  }
+
+  getShoppingCartItems(): ShoppingCartItem[] {
+    return this.shoppingCart;
+  }
+
+
+  placeOrder(): Observable<{}> {
+    const order = new OrderInput();
+    order.customer = 'doej';
+    order.products = [];
+    for (const item of this.shoppingCart) {
+      order.products.push({productId: item.product.id, quantity: item.quantity});
+    }
+    this.shoppingCart = [];
```

Comment:
I guess you should clear the cart only after you get the response for the create order (if there is an error, you should not clear it).